contracts/cw-core-factory/src/tests.rs

```diff
@@ -0,0 +1,86 @@
+use cosmwasm_std::{to_binary, Addr, Empty};
+use cw_core::msg::{Admin, ModuleInstantiateInfo};
+use cw_multi_test::{App, Contract, ContractWrapper, Executor};
+
+use crate::msg::{ExecuteMsg, InstantiateMsg};
+
+fn factory_contract() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        crate::contract::execute,
+        crate::contract::instantiate,
+        crate::contract::query,
+    )
+    .with_reply(crate::contract::reply);
+    Box::new(contract)
+}
+
+fn cw_core_contract() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        cw_core::contract::execute,
+        cw_core::contract::instantiate,
+        cw_core::contract::query,
+    )
+    .with_reply(crate::contract::reply)
+    .with_migrate(cw_core::contract::migrate);
+    Box::new(contract)
+}
+
+#[test]
+pub fn test_set_admin() {
+    let mut app = App::default();
+    let code_id = app.store_code(factory_contract());
+    let instantiate = InstantiateMsg {};
+    let factory_addr = app
+        .instantiate_contract(
+            code_id,
+            Addr::unchecked("CREATOR"),
+            &instantiate,
+            &[],
+            "cw-core-factory",
+            None,
+        )
+        .unwrap();
+
+    // Instantiate core contract using factory.
+    let cw_core_code_id = app.store_code(cw_core_contract());
+    let instantiate_core = cw_core::msg::InstantiateMsg {
+        admin: None,
+        name: "DAO DAO".to_string(),
+        description: "A DAO that builds DAOs.".to_string(),
+        image_url: None,
+        automatically_add_cw20s: true,
+        automatically_add_cw721s: true,
+        voting_module_instantiate_info: ModuleInstantiateInfo {
+            code_id,
+            msg: to_binary(&instantiate).unwrap(),
+            admin: Admin::CoreContract {},
+            label: "voting module".to_string(),
+        },
+        proposal_modules_instantiate_info: vec![ModuleInstantiateInfo {
+            code_id,
+            msg: to_binary(&instantiate).unwrap(),
+            admin: Admin::CoreContract {},
+            label: "voting module".to_string(),
+        }],
+        initial_items: None,
+    };
+
+    // multi-test does not support UpdateAdmin yet :(
```

Comment:
This is unfortunate. Unsure what to do about this other than relying on manual testing until the support is added.

Reply:
Made an issue in cw-plus: https://github.com/CosmWasm/cw-plus/issues/744

Reply:
thank you!